pmm-app/src/pmm-inventory/DataService.ts

```diff
@@ -0,0 +1,88 @@
+/* tslint:disable */
+export class CustomLabelsModel {
+  key: string;
+  value: string;
+
+  constructor(customLabel) {
+    this.key = customLabel[0];
+    this.value = customLabel[1];
+  }
+}
+
+export class CommonModel {
+  custom_labels: CustomLabelsModel[];
+  agentType: string;
+  isDeleted: boolean;
+
+  constructor(params, type) {
+    const { custom_labels, ...rest } = params;
+    this.custom_labels =
+      custom_labels && Object.keys(custom_labels).length ? Object.entries(custom_labels).map(item => new CustomLabelsModel(item)) : [];
+    this.agentType = type;
+    this.isDeleted = false;
+    Object.keys(rest).forEach(param => {
+      this[param] = rest[param];
+    });
+  }
+}
+
+export class InventoryDataService {
+  constructor() {}
+
+  static generateStructure(item) {
+    const addAgentType = Object.keys(item).map(agentType => new Object({ agentType: agentType, params: item[agentType] }));
+    const createParams = addAgentType.map(agent =>
+      agent['params'].map(arrItem => new CommonModel(arrItem, InventoryDataService.getType(agent['agentType'])))
+    );
+    return [].concat(...createParams);
+  }
+
+  static getType(type) {
+    switch (type) {
```

Comment:
@lunaticusgreen, what do you think to change this `switch/case` to the next form:

```suggestion
    typeMap = {
       external_exporter: 'External exporter',
       mongodb_exporter: 'MongoDB exporter'
       ....
    }
    return typeMap[type] || ''
```

I know, it is a more pythonic way, but at least for me looks more compact and readable.

Reply:
Well, pythonic sounds good enough for me :D 
Switch here is legacy after migration this code from qan-app and simplification, I thought that we may need it in case if we need to add some additional logic here, but I think you're right, we just can add it if we need it